src/MCPClient/lib/clientScripts/verify_and_restructure_transfer_bag.py

```diff
@@ -55,6 +56,21 @@ def restructureBagForComplianceFileUUIDsAssigned(
         dirPath = os.path.join(unitPath, dir)
         dirDataPath = os.path.join(unitPath, "data", dir)
         if os.path.isdir(dirDataPath):
+            # When moving metadata directory from the bag, check if
+            # there is already a metadata dir at the top level
+            # (it could have been created by automation tools)
+            # (ref. AM issue 1022 )
+            topleveldir_exists = False
+            if dir == "metadata" and os.path.isdir(dirPath):
+                # temporarily rename top level metadata directory
+                job.pyprint(
+                    "{}: temporarily renaming pre-existing directory".format(dir)
+                )
+                topleveldir_exists = True
+                tempdir = "{}_{}".format(dir, os.getpid())
```

Comment:
I like this use of `getpid()` to create something unique and simple! I think given the opportunity to refactor this, I'd avoid creating `tempdir` and call `"...".format(...)` directly on line 71 to tighten up on lines and avoid single use vars.

Reply:
done!